### PR TITLE
[Xedra Evolved] Eater Sixth Sense

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Give Eater an advanced ability once they've gained some experience in the Class.

#### Describe the solution
Adds a sixth sense enchantment to the Eater class in order to sense creatures of the Nether.  I'll see if I can get someone to add a spell school count condition and if not I'll comment it out and add a spell that you learn upon reaching any Eater spell level of 15.   That spell will then give you the trait that powers the enchantment.

#### Describe alternatives you've considered
None

#### Testing
Tested and worked well.

### CODE REVIEW

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 1]
```diff
@@ -31,5 +31,25 @@
...
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_eater_escalation",
+    "recurrence": 1,
+    "condition": { "compare_num": [ { "u_val": "spell_count", "school": "EATER" }, "=", { "const": 6 } ] },
+    "effect": [
+      {
+        "u_message": "You suddenly realize that your ability to sense the unnatural has expanded to a level beyond those of the normal five senses.",
+        "type": "good",
+        "u_add_trait": "EATER_SIXTH_SENSE"
```
Comment (dseguin) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:51]:
```suggestion
      },
      {
        "u_add_trait": "EATER_SIXTH_SENSE"
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 2]
```diff
@@ -31,5 +31,27 @@
...
+      {
+        "u_message": "You suddenly realize that your ability to sense the unnatural has expanded to a level beyond those of the normal five senses.",
+        "type": "good",
+      },
+      {
+        "u_add_trait": "EATER_SIXTH_SENSE"
+      }
+    ]
   }
 ]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:57]:
```suggestion
]

```

File: data/mods/Xedra_Evolved/effects/effects.json [hunk 3]
```diff
@@ -24,6 +24,20 @@
...
+    "type": "effect_type",
+    "id": "sense_unnatural",
+    "name": [ "Sense Unnatural" ],
+    "desc": [ "You can sense the presence of things that do not belong in this world." ],
+    "apply_message": "",
+    "remove_message": "Your awareness of the unnatural fades.",
+    "rating": "good",
+    "enchantments": [
+      {
+        "values": [ { "value": "SIGHT_RANGE_NETHER", "add": { "math": [ "(( u_val('spell_level', , 'school: EATER') * 1.5) + 3)" ] } } ]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effects.json:37]:
```suggestion
        "values": [ { "value": "SIGHT_RANGE_NETHER", "add": { "math": [ "(( u_val('spell_level', 'school: EATER') * 1.5) + 3)" ] } } ]
```

**Comment by Maleclypse** [comment]
@Standing-Storm do you see what I've done wrong here?

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 4]
```diff
@@ -31,5 +31,25 @@
...
+    "id": "EOC_EATER_CHECK",
+    "recurrence": [ { "global_val": "ps_min_woc", "default": "1 days" }, { "global_val": "ps_max_woc", "default": "2 days" } ],
+    "global": true,
+    "effect": { "run_eocs": "EOC_eater_escalation" }
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_eater_escalation",
+    "recurrence": 1,
+    "condition": { "compare_num": [ { "u_val": "spell_count", "school": "EATER" }, "=", { "const": 6 } ] },
```
Comment (Standing-Storm) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:46]: > "condition": { "compare_num": [ { "u_val": "spell_count", "school": "EATER" }, "=", { "const": 6 } ] },

"condition": { "compare_num": [ { "u_val": "spell_count", "school": "EATER" }, ">=", { "const": 6 } ] },

File: data/mods/Xedra_Evolved/enchantments/mutations.json [hunk 5]
```diff
@@ -25,5 +25,21 @@
...
+    "condition": "ALWAYS",
+    "hit_you_effect": [ { "id": "eater_sixth_sense", "hit_self": true, "once_in": 12 } ]
+  },
+  {
+    "id": "eater_sixth_sense",
+    "type": "SPELL",
+    "name": "Eater Sixth Sense",
+    "description": "Heals the caster.",
+    "valid_targets": [ "self" ],
+    "effect": "sense_unnatural",
```
Comment (Standing-Storm) [inline data/mods/Xedra_Evolved/enchantments/mutations.json:41]: > "effect": "sense_unnatural",
"effect": "attack",
"effect_str": "sense_unnatural",

Needs a duration as well.

**Review by Standing-Storm — CHANGES_REQUESTED** [review]

File: data/mods/Xedra_Evolved/enchantments/mutations.json (hunk 5 above)
Comment (Standing-Storm) [reply data/mods/Xedra_Evolved/enchantments/mutations.json:41]: I don't know how to do the fancy Github changes form so I just used quotes. 🥴

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json (hunk 4 above)
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:46]:
```suggestion
    { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
```

File: data/mods/Xedra_Evolved/enchantments/mutations.json (hunk 5 above)
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/enchantments/mutations.json:41]:
```suggestion
    "effect": "attack",
    "effect_str": "sense_unnatural",
    "duration": 86000
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 6]
```diff
@@ -43,7 +43,7 @@
     "type": "effect_on_condition",
     "id": "EOC_eater_escalation",
     "recurrence": 1,
-    "condition": { "compare_num": [ { "u_val": "spell_count", "school": "EATER" }, "=", { "const": 6 } ] },
+    { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
```
Comment (Standing-Storm) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:46]:
```suggestion
    "condition": { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 7]
```diff
@@ -31,5 +31,25 @@
...
+    "id": "EOC_EATER_CHECK",
+    "recurrence": [ { "global_val": "ps_min_woc", "default": "1 days" }, { "global_val": "ps_max_woc", "default": "2 days" } ],
+    "global": true,
+    "effect": { "run_eocs": "EOC_eater_escalation" }
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_eater_escalation",
+    "recurrence": 1,
+    "condition": { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:46]:
```suggestion
    "condition": { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] },
```

File: data/mods/Xedra_Evolved/enchantments/mutations.json [hunk 8]
```diff
@@ -25,5 +25,11 @@
...
     "condition": "ALWAYS",
     "values": [ { "value": "MAX_MANA", "add": { "math": [ "u_skill('deduction')*50" ] } } ]
-  }
+  },
+    {
+    "type": "enchantment",
+    "id": "EATER_SIXTH_SENSE",
+    "condition": "ALWAYS",
+    "values": [ { "value": "MAX_MANA", "add": { "math": [ "u_skill('deduction')*50" ] } } ]
+  },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/enchantments/mutations.json:34]:
```suggestion
  }
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 9]
```diff
@@ -31,5 +31,24 @@
...
+    "id": "EOC_EATER_CHECK",
+    "global": true,
+    "recurrence": [ "12 hours", "72 hours" ],
+    "condition": {
+      "and": [
+        { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] },
+        { "not": [ { "u_has_trait": "EATER_SIXTH_SENSE" } ] }
+      ]
+    },
+    "effect": [
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:46]:
```suggestion
    "deactivate_condition": { "u_has_trait": "EATER_SIXTH_SENSE" },
    "effect": [
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 10]
```diff
@@ -31,5 +31,24 @@
...
     "condition": { "u_has_items": { "item": "book_teens_deduction", "count": 1 } },
     "effect": [ { "u_message": "XE_SCRAGGY-BOO_MESSAGES_1", "snippet": true } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_EATER_CHECK",
+    "global": true,
+    "recurrence": [ "12 hours", "72 hours" ],
+    "condition": {
+      "and": [
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:41]:
```suggestion
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 11]
```diff
@@ -31,5 +31,24 @@
...
+        { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] },
+        { "not": [ { "u_has_trait": "EATER_SIXTH_SENSE" } ] }
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:43]:
```suggestion
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 12]
```diff
@@ -31,5 +31,24 @@
...
     "effect": [ { "u_message": "XE_SCRAGGY-BOO_MESSAGES_1", "snippet": true } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_EATER_CHECK",
+    "global": true,
+    "recurrence": [ "12 hours", "72 hours" ],
+    "condition": {
+      "and": [
+        { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:42]:
```suggestion
        { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 13]
```diff
@@ -31,5 +31,23 @@
...
     "effect": [ { "u_message": "XE_SCRAGGY-BOO_MESSAGES_1", "snippet": true } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_EATER_CHECK",
+    "global": true,
+    "recurrence": [ "12 hours", "72 hours" ],
+    "condition": {
+        { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
+      ]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:42]:
```suggestion
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 14]
```diff
@@ -31,5 +31,22 @@
     "recurrence": [ "72 hours", "168 hours" ],
     "condition": { "u_has_items": { "item": "book_teens_deduction", "count": 1 } },
     "effect": [ { "u_message": "XE_SCRAGGY-BOO_MESSAGES_1", "snippet": true } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_EATER_CHECK",
+    "global": true,
+    "recurrence": [ "12 hours", "72 hours" ],
+    "condition": {
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:40]:
```suggestion
    "condition": 
```

File: data/mods/Xedra_Evolved/effects/effect_on_condition.json [hunk 15]
```diff
@@ -31,5 +31,22 @@
...
+        { "math": [ "u_val('spell_count', 'school: EATER')", ">=", "6" ] }
+    },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effect_on_condition.json:42]:
```suggestion
```

**Pull request merged**